core-rust/core-api-server/src/core_api/handlers/lts/state_account_all_fungible_resource_balances.rs

```diff
@@ -0,0 +1,84 @@
+use crate::core_api::*;
+use radix_engine::types::{Decimal, IndexMap};
+use state_manager::store::traits::QueryableProofStore;
+use state_manager::{
+    jni::state_manager::ActualStateManager,
+    query::{dump_component_state, VaultData},
+};
+
+#[tracing::instrument(skip(state), err(Debug))]
+pub(crate) async fn handle_rc_state_account_all_fungible_resource_balances(
+    state: State<CoreApiState>,
+    request: Json<models::LtsStateAccountAllFungibleResourceBalancesRequest>,
+) -> Result<Json<models::LtsStateAccountAllFungibleResourceBalancesResponse>, ResponseError<()>> {
+    core_api_read_handler(
+        state,
+        request,
+        handle_rc_state_account_all_fungible_resource_balances_internal,
+    )
+}
+
+fn handle_rc_state_account_all_fungible_resource_balances_internal(
+    state_manager: &ActualStateManager,
+    request: models::LtsStateAccountAllFungibleResourceBalancesRequest,
+) -> Result<models::LtsStateAccountAllFungibleResourceBalancesResponse, ResponseError<()>> {
+    assert_matching_network(&request.network, &state_manager.network)?;
+
+    if !request.account_address.starts_with("account_") {
+        return Err(client_error(
+            "Only component addresses starting with account_ currently work with this endpoint.",
+        ));
+    }
+
+    let mapping_context = MappingContext::new(&state_manager.network);
+    let extraction_context = ExtractionContext::new(&state_manager.network);
+
+    let component_address =
+        extract_component_address(&extraction_context, &request.account_address)
+            .map_err(|err| err.into_response_error("account_address"))?;
+
+    let component_dump = dump_component_state(state_manager.store(), component_address)
```

Comment:
If the component doesn't exist, what error do we get here?

We should ensure that:
* If the account address is a _normal_ account address, then we return a 404
* If the account address is a _virtual_ account address, then we return a response with no resources.

_Note:_ We can tell if the account address is virtual or not by matching on the component address